DSpace/dspace-angular
# Fix DSO 404 pages

## References
This PR closes #865 

## Description
This PR refactors the resolve method of `DSOBreadcrumbResolver` to let RemoteData objects through containing an error response. This avoids the resolver blocking 404 pages from loading.

## Instructions for Reviewers
Changes made:
- Replaced `getSucceededRemoteData()` with `filter((rd) => hasValue(rd.error) || hasValue(rd.payload))` and `take(1)`
- Modified the map in this pipe to return undefined if our payload is empty

How to test:
- Visit a page for a DSpaceObject with a UUID that doesn't exist (e.g. community/collection/item pages). The page should display a 404 instead of an infinite loading indicator.
- Visit a page for an existing DSpaceObject and make sure the changes haven't broken any of the existing functionalities (mostly the breadcrumbs)

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
